common/src/main/java/net/opentsdb/stats/Trace.java

```diff
@@ -24,13 +24,44 @@
 public interface Trace {
 
   /**
-   * Returns a new span builder for this tracer.
+   * Returns a new span builder for this tracer without any tags.
    * @param id A non-null and non-empty span ID.
    * @return A non-null span builder.
+   * @throws IllegalArgumentException if a required argument was invalid.
    */
   public SpanBuilder newSpan(final String id);
   
   /**
+   * Returns a new span builder for this tracer with the given tags set.
+   * @param id A non-null and non-empty span ID.
+   * @param tags A list of key/value pairs. Must be an even number of non-null
```

Comment:
Nope, just here though I may do it elsewhere. It's just a shortcut to setting tags quicker. Order does matter in k/v pairs, but other than that doesn't matter. I guess I could build it out as a proper builder.

Reply:
I raised the issue only because it's a point of contention, generally. I feel happy with either solution so long as it's reasonably consistent. I can think of someone who'd applaud you for minimizing object creation; using a builder would merely add more objects and code verbosity.